yb-voyager/src/queryparser/traversal_plpgsql.go

```diff
@@ -228,3 +216,183 @@ func formatExprQuery(q string) string {
+    LANGUAGE plpgsql
+    AS $$
+DECLARE
+    emp_salary employees.salary%TYPE;  
+BEGIN
+    SELECT salary INTO emp_salary
+    FROM employees
+    WHERE employee_id = emp_id;
+    RETURN emp_salary;
+END;
+$$;
+[
+    {
+        "PLpgSQL_function": {
+            "datums": [
+                {
+                    "PLpgSQL_var": {
+                        "refname": "emp_id",
+                        "datatype": {
+                            "PLpgSQL_type": {
+                                "typname": "UNKNOWN" 
+                            }
+                        }
+                    }
+                },
+                {
+                    "PLpgSQL_var": {
+                        "refname": "found",
+                        "datatype": {
+                            "PLpgSQL_type": {
+                                "typname": "UNKNOWN"
+                            }
+                        }
+                    }
+                },
+                {
+                    "PLpgSQL_var": {
+                        "refname": "emp_salary",
+                        "lineno": 3,
+                        "datatype": {
+                            "PLpgSQL_type": {
+                                "typname": "employees.salary%TYPE"
+                            }
+                        }
+                    }
+                },
+                {
+                    "PLpgSQL_row": {
+                        "refname": "(unnamed row)",
+                        "lineno": 5,
+                        "fields": [
+                            {
+                                "name": "emp_salary",
+                                "varno": 2
+                            }
+                        ]
+                    }
+                }
+            ],"action": {
+               ....
+            }
+        }
+    },
+
+	Caveats:
+	1. Not returning typename for variables in function parameter from this function (in correct in json as UNKNOWN), for that using the GetTypeNamesFromFuncParameters()
+	2. Not returning the return type from this function (not available in json), for that using the GetReturnTypeOfFunc()
+*/
+func GetAllTypeNamesInPlpgSQLStmt(query string) ([]string, error) {
+	parsedJson, parsedJsonMap, err := getParsedJsonMap(query)
+	if err != nil {
+		return []string{}, nil
+	}
+	function := parsedJsonMap[PLPGSQL_FUNCTION]
+	parsedFunctionMap, ok := function.(map[string]interface{})
+	if !ok {
+		return []string{}, fmt.Errorf("error getting the PlPgSQL_Function field in parsed json-%s", parsedJson)
+	}
+
+	datums := parsedFunctionMap[DATUMS]
+	datumList, isList := datums.([]interface{})
+	if !isList {
+		return []string{}, fmt.Errorf("error getting type names datums field is not list in parsed json-%s", parsedJson)
+	}
+
+	var typeNames []string
+	for _, datum := range datumList {
+		datumMap, ok := datum.(map[string]interface{})
+		if !ok {
```

Comment:
do we know in what situations it would NOT be a map[string]interface. If not, should we error out? 
If we do not want to error and just move on, atleast, let's log.ERROR?